cytounet/model.py

```diff
@@ -25,63 +24,27 @@ def dice_coef(y_true, y_pred, smooth=1):
 
 def dice_coef_loss(y_true, y_pred):
     """
-
-       :param y_true: Train ground truth
-       :param y_pred: Predicted
-       :return: Returns dice loss
-
-       """
-    return 1 - dice_coef(y_true, y_pred)
-
-
-# bias variance tradeoff
-# https://blog.insightdatascience.com/bias-variance-tradeoff-explained-fa2bc28174c4
-def unet_simple(pretrained_weights=None, metrics=None, input_size=(256, 256, 1), optimiser="Adam",
-                learning_rate=3e-6, loss="binary_crossentropy", model_name="unet_simple"):
-    """
-    :param pretrained_weights: If a pretrained model exists, provide it here for fine tuning
-    :param input_size: size of the first layer(input_layer). Defaults to (256, 256, 1)
-    :param optimiser: Optimiser to use. One of SGD or Adam. Defaults to Adam.
-    :param learning_rate: Learning rate to use with the Adam optimiser. Defaults to 3e-6
-    :param loss:  Loss function to use. Defaults to binary_crossentropy
-    :param metrics: Metrics to use. Defaults to accuracy
-    :param model_name Name of the model, defaults to unet_simple
-    :return: A simple(r) unet model.
+    :param y_true: Train ground truth
+    :param y_pred: Predicted
+    :return: Returns dice loss
     """
-    if metrics is None:
-        metrics = ['accuracy']
-    inputs = Input(shape=input_size)
-    conv1 = Conv2D(64, 3, activation="relu", padding="same", kernel_initializer="he_normal")(inputs)
-    conv1 = BatchNormalization()(conv1)
-    conv1 = Conv2D(64, 3, activation="relu", padding="same", kernel_initializer="he_normal")(conv1)
-
-    conv2 = Conv2D(128, 3, activation="relu", padding="same", kernel_initializer='he_normal')(conv1)
-    conv2 = BatchNormalization()(conv2)
-    conv2 = Conv2D(128, 3, activation="relu", padding="same", kernel_initializer='he_normal')(conv2)
-    conv2 = BatchNormalization()(conv2)
-
-    up3 = Conv2D(64, 2, activation="relu", padding="same", kernel_initializer="he_normal")(conv2)
-    merge3 = concatenate([up3, conv1], axis=3)
-    conv4 = Conv2D(1, 1, activation="relu")(merge3)
-
-    model = Model(inputs=inputs, outputs=conv4, name=model_name)
-
-    optimiser_list = {'Adam': Adam(lr=learning_rate),
-                      'SGD': SGD(learning_rate=learning_rate, momentum=0.99)}
-
-    model.compile(optimizer=optimiser_list[optimiser],
-                  loss=loss, metrics=metrics)
-    # model.name = model_name
+    return 1 - dice_coef(y_true, y_pred)
 
-    if pretrained_weights is not None:
-        model.load_weights(pretrained_weights)
 
-    return model
+def apply_conv(input_tile, filters, kernel_regularizer, layered_conv: bool):
```

Comment:
Thanks for this, I have tried running but the results are not pretty. Something I did not understand from the code is the `return Conv2D` if double layered. Did you intend to set:

```python
if layered_conv:
    result = Conv2D(filters, 3, activation='relu', kernel_initializer='he_normal', padding='same',
                      kernel_regularizer=kernel_regularizer)(result)
```

Otherwise it seems now for the `double_layered` it returns one non batch normalized `Conv2D` layer.

Reply:
Read the following comment after you read my comment to lines 38-40:
Maybe I shouldn't have applied batch_normalization if 'layered_conv' is False (simple model)?

Reply:
I think it's fine to apply a `BatchNorm` there.